### PR TITLE
Transcendental functions

### PR DESCRIPTION
**Pull request by buggymcbugfix**
This PR will add support for exponentiation, logarithms, etc.

### CODE REVIEW

File: tests/script/negative/arithmetic/precision-in-global-def.out [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Error at Line 6, Column 8:
```
Comment (faineance) [inline tests/script/negative/arithmetic/precision-in-global-def.out]: might be worth adding some negatives cases to test the `LogOfNegative` error

File: src/Language/FCL/Typecheck.hs [hunk 2]
```diff
@@ -503,19 +503,13 @@ tcDefn def = extendContextInferM Global =<< case def of
...
     -- Check if the stated definition type and the rhs expr type match
-    unifyDef :: Name -> LExpr -> TypeInfo -> TypeInfo -> Either TypeErrInfo ()
-    unifyDef nm le t1 t2 = do
-      case runSolverM emptyInferState{ constraints = [Constraint (Just le) t1 t2] } of
-        Right _ -> pure ()
-        Left (terr :| _) ->
-          case errInfo terr of
-            UnificationFail ti1 ti2 -> Left $ InvalidDefinition nm (locVal le) (ttype ti1) (ttype ti2)
-            otherwise               -> panic "Solver should fail with UnificationFail"
+    unifyDef nm le t1 t2
```
Comment (faineance) [inline src/Language/FCL/Typecheck.hs]: Did we intend to delete the type annotation for this function?

**Comment by Acentelles** [comment]
We are defaulting to 9 decimal places when doing exponentiation. Is it enough? This can be increased. It is a business question

File: package.yaml [hunk 3]
```diff
@@ -102,6 +102,9 @@ library:
   - asn1-types
   - asn1-encoding
 
+  # Numeric
+  - numeric-quest
```
Comment (sdiehl) [inline package.yaml]: I don't like being the only library in the ecosystem to use this unmaintained package.

**Pull request merged**